### ISSUE TITLE
TS2304: Cannot find name 'T' with JSDoc

### ISSUE THREAD

**Issue by jbrimeyer** [microsoft/typescript-go#2919]
In a JavaScript file `index.js`:
```javascript
/** @template {{done(): void}} T */
class Foo {
	/** @typedef {(t: T) => void} Bar */

	/** @type {Map<string, Map<string, Bar>>} */
	baz = new Map();
}
```

## With TypeScript Version 5.9.3

```
npx tsc --checkJs --strict --noEmit --lib es2021 index.js
```

No errors reported

## With tsgo Version 7.0.0-dev.20260226.1

```
npx tsgo --checkJs --strict --noEmit --lib es2021 index.js
```

Error TS2304 is generated:

```
index.js:4:20 - error TS2304: Cannot find name 'T'.

4  /** @typedef {(t: T) => void} Bar */
                     ~


Found 1 error in index.js:4
```

**Comment by jakebailey** [comment]
I don't think this worked <=6.0 how you thought it did. Look at: [Playground Link](https://www.typescriptlang.org/play/?filetype=js#code/PQKhAIAEBcFMFsAOAbAhncBvTATA9gHawAUAlAFzgBueAljgL4PgAq4IwAsAFADGaAZwHgAYnjxYeASFAQYAT0SwcsAGZZi0Si1LgAvAD5qdRuABCqAE7su3abKjRFsLAFlUiADwDol2gQBzABpwdy8fP0CQi0sDA2YOaQAjVAAvfXAiAHdQjzIAbh4GIA)

This is emitted in `d.ts` as:

```ts
/** @template {{done(): void}} T */
declare class Foo<T extends {
    done(): void;
}> {
    /** @typedef {(t: T) => void} Bar */
    /** @type {Map<string, Map<string, Bar>>} */
    baz: Map<string, Map<string, (t: T) => void>>;
}
type Bar = (t: T) => void;
```

Note that `Bar` is hoisted (types cannot be defined within a class!), and then `T` now points to nothing. I think the new code is now detecting this, because we actually reparse the JS typedef into the equivalent TS, whereas the original code I think was silently broken?
